src/index.js

```diff
@@ -145,9 +145,8 @@ export function initialize(env, user, specifiedOptions, platform, extraOptionDef
   function sendFlagEvent(key, detail, defaultValue, includeReason) {
     const user = ident.getUser();
     const now = new Date();
-    const value = detail ? detail.value : null;
     if (!options.allowFrequentDuplicateEvents) {
-      const cacheKey = JSON.stringify(value) + (user && user.key ? user.key : '') + key; // see below
+      const cacheKey = JSON.stringify(detail) + JSON.stringify(user) + key; // see below
```

Comment:
Perhaps only apply these changes if `includeReason` or `inlineUsersInEvents` is true.

Reply:
I have a different concern: the behavior of `JSON.stringify` is not deterministic such that serializing an object with the same properties would always serialize them in the same order. That was already a flaw in the existing logic since `value` could be a JSON object, but that's a rare enough usage that it hasn't been an issue before, especially since the value of `value` is always (for any valid flag) an exact copy of the same internal flag value. It's more problematic with regard to the user properties, and the `detail` object which could be constructed in a different order in different places in the code. In general it's not ideal to use a serialized complex object as a cache key if it doesn't have a canonical encoding.

Serializing the entire user object for every evaluation also adds potentially non-trivial overhead to applications that do frequent evaluations (with even more overhead if we add logic to canonicalize objects as mentioned above), so at a minimum this changed behavior would need to be opt-in.